Added feature description for AWS ci setup (#6)
exasol/exasol-test-setup-abstraction-java: ci-account-setup/doc/features.md

> # Features of th AWS CI Account Setup

Reviewer: ```suggestion
# Features of the AWS CI Account Setup
```

The AWS CI Account Setup provisions an AWS account for running tests from CI securely.

## Create AWS resources from GitHub Actions

`feat~create-aws-cresources-in-ci~1`

> We can create AWS resources from CI builds in out GitHub Actions CI builds.

Reviewer: ```suggestion
The AWS CI Account Setup can create AWS resources from CI builds in out GitHub Actions CI builds.
```

Author: The AWS CI Account Setup creates a user for CI. With the user we can create AWS resources from CI builds in our GitHub Actions CI builds.

Rationale:

* We can not use a different CI system since [Release Droid](https://github.com/exasol/release-droid) is built using GitHub actions.
* Even so it's favorable to use local resources like local versions in docker containers for some integrations we need to test against the real cloud services. Possibles reasons are:
    * There is no adequate local version
    * Performance testing

Needs: dsn

## Damage Minimization on Hacking Attacks

`feat~damage-minimization-on-hacking-attacks~1`

> The AWS CI account setup implements measurements to reduce the damage (costs, abuse) in case an attacker gains access to the CI account.

Reviewer: ```suggestion
The AWS CI Account Setup implements measurements to reduce the damage (costs, abuse) in case an attacker gains access to the CI account.
```

> The AWS CI account setup implements measurements to reduce the damage (costs, abuse) in case an attacker gains access to the CI account.

Reviewer: this seems quite vague, and doesn't say much. What is this feature about?

Author: Basically that we will limit the account as much as possible.
I will precise in the requirements.

Rationale:
Since the credentials are exposed to the CI there is an increased risk that an attacker could gain access to them. For that case the damage should be as small as possible.

Needs: dsn

> ## Prevention of Costs Caused by CI Builds That Failed to Clean up

Reviewer: ```suggestion
## Prevention of Costs Caused by CI Builds That Failed to Clean Up
```

Author: according to the Associated Press style it must be in lower case

`feat~clean-up~1`

> The AWS CI account setup handles resources that were created but not deleted by the CI.

Reviewer: ```suggestion
The AWS CI Account Setup handles resources that were created but not deleted by the CI.
```
